frontend/src/views/cronjob/helper.ts

```diff
@@ -164,50 +164,26 @@ export function transSpecToObj(spec: string) {
 
 export function transSpecToStr(spec: string): string {
     const specObj = transSpecToObj(spec);
-    let str = '';
-    if (specObj.specType.indexOf('N') === -1 || specObj.specType === 'perWeek') {
-        str += i18n.global.t('cronjob.' + specObj.specType) + ' ';
-    } else {
-        str += i18n.global.t('cronjob.per') + ' ';
-    }
     switch (specObj.specType) {
         case 'perMonth':
-            str +=
-                specObj.day +
-                i18n.global.t('cronjob.day') +
-                ' ' +
-                loadZero(specObj.hour) +
-                ':' +
-                loadZero(specObj.minute);
-            break;
+            return i18n.global.t('cronjob.perMonthHelper', [specObj.day, specObj.hour, loadZero(specObj.minute)]);
         case 'perWeek':
-            str += loadWeek(specObj.week) + ' ' + loadZero(specObj.hour) + ':' + loadZero(specObj.minute);
-            break;
+            return i18n.global.t('cronjob.perWeekHelper', [
+                loadWeek(specObj.week),
+                specObj.hour,
+                loadZero(specObj.minute),
+            ]);
         case 'perDay':
-            str += loadZero(specObj.hour) + ':' + loadZero(specObj.minute);
-            break;
+            return i18n.global.t('cronjob.perDayHelper', [specObj.hour, loadZero(specObj.minute)]);
+        case 'perHour':
+            return i18n.global.t('cronjob.perHourHelper', [loadZero(specObj.minute)]);
         case 'perNDay':
-            str +=
-                specObj.day +
-                i18n.global.t('commons.units.day') +
-                ', ' +
-                loadZero(specObj.hour) +
-                ':' +
-                loadZero(specObj.minute);
-            break;
+            return i18n.global.t('cronjob.perNDayHelper', [specObj.day, specObj.hour, loadZero(specObj.minute)]);
         case 'perNHour':
-            str += specObj.hour + i18n.global.t('commons.units.hour') + ', ' + loadZero(specObj.minute);
-            break;
-        case 'perHour':
-            str += loadZero(specObj.minute);
-            break;
+            return i18n.global.t('cronjob.perNHourHelper', [specObj.hour, loadZero(specObj.minute)]);
         case 'perNMinute':
-            str += loadZero(specObj.minute) + i18n.global.t('commons.units.minute');
-            break;
+            return i18n.global.t('cronjob.perNMinuteHelper', [loadZero(specObj.minute)]);
         case 'perNSecond':
-            str += loadZero(specObj.second) + i18n.global.t('commons.units.second');
-            break;
+            return i18n.global.t('cronjob.perNSecondHelper', [loadZero(specObj.second)]);
     }
-
-    return str + ' ' + i18n.global.t('cronjob.handle');
 }
```

Comment:
The `transSpecToStr` function has several issues and can be optimized:

**Issues:**

1. **Irregular Control Structures:** The function uses multiple control structures (`if`, `switch`, etc.) with complex logic inside them. These can make the code harder to debug and maintain.

2. **Return Statements Early:** After returning from each case block, there's no need for additional return statements. This can lead to unnecessary lines of code.

3. **String Concatenation:** While not a significant issue, using template literals (`\`\``) instead of string concatenation might improve readability and performance slightly.

4. **Magic Numbers:** Using `i18n.global.t(...)` directly without any explanation is confusing to others reading the code.

5. **Duplicate Code:** Similar constructs are repeated in different cases, which suggests that some common functionality could be extracted into helper functions.

6. **Potential Edge Cases Missing:** There are no checks for missing fields or invalid inputs, though this isn't explicitly stated in your input text.

7. **Missing Localization Keys:** Some localization keys (e.g., `'cronjob.'`) aren't present based on your comment about it being loaded dynamically. Ensure they exist.

### Optimized Solution:

Here's a cleaned-up version of the `transSpecToStr` function with better structure and comments:

```javascript
function transSpecToStr(spec: string): string {
    const specObj = transSpecToObj(spec);

    // Create an array to hold pieces of the result message
    const parts: string[] = [];

    // Convert day part
    if (specObj.specType !== 'Nday' && specObj.specType !== 'Nhour') {
        parts.push(i18n.global.t('cronjob.' + specObj.specType));
    }

    switch (specObj.specType) {
        case 'perMonth':
            parts.push(
                `${specObj.day} ${i18n.global.t('cronjob.day')}` +
                ` ${loadZero(specObj.hour)}:${loadZero(specObj.minute)}`
            );
            break;

        // ... repeat similar patterns but with unique helper functions

        default:
            console.error(`Unrecognized spec type: ${specObj.specType}`);
    }

    // Join all parts with spaces and append a handle
    return parts.join(" ") + " " + i18n.global.t('cronjob.handle');
}
```

In summary, the main changes include:
- Consolidating control structures.
- Removing redundant return statements.
- Improving code readability by extracting shared logic.
- Adding comments where appropriate to explain decisions.

This should make the function easier to understand and maintain while also adhering to best practices.